Assets/Mirror/Components/NetworkTransformBase.cs

```diff
@@ -16,131 +16,265 @@
         {
             public float timeStamp;
-            // use local position/rotation for VR support
             public Vector3 localPosition;
             public Quaternion localRotation;
             public Vector3 localScale;
             public float movementSpeed;
+
+            public bool isValid => timeStamp != 0;
         }
+
+        // Is this a client with authority over this transform?
+        // This component could be on the player object or any object that has been assigned authority to this client.
+        bool IsOwnerWithClientAuthority => hasAuthority && clientAuthority;
+
         // interpolation start and goal
-        DataPoint start;
-        DataPoint goal;
+        public DataPoint start = new DataPoint();
+        public DataPoint goal = new DataPoint();
 
-        // local authority send time
-        float lastClientSendTime;
+        // We need to store this locally on the server so clients can't request Authority when ever they like
+        bool clientAuthorityBeforeTeleport;
 
-        // serialization is needed by OnSerialize and by manual sending from authority
-        // public only for tests
-        public static void SerializeIntoWriter(NetworkWriter writer, Vector3 position, Quaternion rotation, Vector3 scale)
+        void FixedUpdate()
         {
-            // serialize position, rotation, scale
-            // note: we do NOT compress rotation.
-            //       we are CPU constrained, not bandwidth constrained.
-            //       the code needs to WORK for the next 5-10 years of development.
-            writer.WriteVector3(position);
-            writer.WriteQuaternion(rotation);
-            writer.WriteVector3(scale);
+            // if server then always sync to others.
+            // let the clients know that this has moved
+            if (isServer && HasEitherMovedRotatedScaled())
+                ServerUpdate();
+
+            if (isClient)
+            {
+                // send to server if we have local authority (and aren't the server)
+                // -> only if connectionToServer has been initialized yet too
+                if (IsOwnerWithClientAuthority)
+                    ClientAuthorityUpdate();
+                else if (goal.isValid)
+                    ClientRemoteUpdate();
+            }
         }
 
-        public override bool OnSerialize(NetworkWriter writer, bool initialState)
+        void ServerUpdate()
         {
-            // use local position/rotation/scale for VR support
-            SerializeIntoWriter(writer, targetComponent.transform.localPosition, targetComponent.transform.localRotation, targetComponent.transform.localScale);
-            return true;
+            if (clientAuthority)
+                RpcMove(targetTransform.localPosition, targetTransform.localRotation, targetTransform.localScale);
+            else
+                RpcForceMove(targetTransform.localPosition, targetTransform.localRotation, targetTransform.localScale);
```

Comment:
what is the need for 2 different Rpcs? I think other clients still need to get the same message even if it is server auth or client auth